cityofaustin/atd-moped
# Re-implement map draw tools

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10403

Adds the draw tools to the components map refactor

Questions/concerns:

- there is some cross-activity when clicking a segment or point from CTN layers near a drawn line or point (capture below). The `clickBuffer` parameter of the draw tools could be updated to avoid this but I also don't want to make it super hard to click drawn features if we drop that too low.
![clickLines](https://user-images.githubusercontent.com/37249039/198742954-1062cd30-970c-4075-aaf6-f76d666ffc39.gif)


## Testing
**URL to test:** 

https://md-10403-draw-tools--atd-moped-main.netlify.app/moped/ or test locally

**Steps to test:**
1. Go to a new or existing project's **Map** tab and then click **New Component**
2. Choose component type **Access Control - Driveway Closure**
3. Zoom until you see the dashed lines of the CTN layer
4. Click the line drawing icon in the top right just above the trash can icon
5. You should see:
   a. the dashed lines of the CTN line layer disappear
   b. the cursor turn into a crosshair
6. Draw a line and complete the line by clicking twice in the last vertex. The vertices of the line should still be circled in green showing that the line is currently active in the draw tools
7. While still active, click the line and hold while dragging it and you should see the location of the drawn line update when you release your click
8. Click away from the line and the green circles should disappear and the dashed CTN line layer should reappear
9. Select a couple of segments from the CTN line layer and click **Save** - you will see the new component shown on the map in magenta
10. Repeat the above steps with the point component type called **Bike Box**
11. While drawing the point component, you should see the features of the first line component shown as muted with a gray color
12. After creating and saving a point type component, highlight each new component by clicking on the boxes in the left sidebar
13. You should see the highlighted component features highlighted in blue on the map and the other components muted with a gray color

---
#### Ship list
